File: backend-ebanking/src/main/java/com/bank/ebanking/model/Operation.java
package com.bank.ebanking.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name="OPERATION")
public @Data class Operation {
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="ID_OPERATION")
	Long id;
	
	@JoinColumn(name="COMPTE_OPERATION")
	@ManyToOne 
	Compte compte;

	
	@Column(name="DATE_OPERATION")
	LocalDateTime date;
	
	@Column(name="SOMME_ESPECE_OPERATION")
	double sommeEspece;
	
	@Column(name="SOMME_COMPTE_OPERATION")
	double sommeCompte;
	
	@Column(name="TYPE_OPERATION")
	String type; // (versement ou retrait  )
	
	@Column(name="DEVISE_OPERATION")
	String devise;
	

}

File: frontend-ebanking/src/app/interfaces/operation.ts
import { Compte } from "./compte";

export class Operation {
    id: number;
    compte: Compte;
    date: Date;
    sommeEspece: number;
    sommeCompte: number;
    type: string;
    devise: string;
    constructor(compte:Compte,sommeEspece:number,type:string){
        this.compte = compte;
        this.sommeEspece = sommeEspece;
        this.type = type;
    }
}